.github/workflows/pipeline-perf-test.yml

```diff
@@ -0,0 +1,38 @@
+name: Pipeline Performance Tests
+
+on:
+    push:
+        branches: [ main ]
+    pull_request:
+        branches: [ main ]
+
+jobs:
+    pipeline-perf-test:
+        if: ${{ contains(github.event.pull_request.labels.*.name, 'performance') }}
+        runs-on: ubuntu-latest
```

Comment:
for now, uses the GitHub CI machine. Will move to OTels dedicated perf test machines once I figure out the steps to onboard to that.
(We were trying it for OTel Rust, and still stuck. Once we fix OTel Rust, I can apply that learning back here)